### ISSUE TITLE
Is it possible to use the linker with any ontology?

### ISSUE THREAD

**Issue by xegulon** [kingsaint/BioMedical-EL#1]
Hi, will it be possible to use the model with any ontology, like MedDRA?

Btw, when will everything be available?

Thanks

**Comment by coltonflowers1** [comment]
You have two options as I see it:

1. Train the model on one of the training datasets mentioned in the paper along with its respective KB, but then use the resulting candidate encoder to encode the entity texts of another similar KB. You'd still have to evaluate a test set linking your documents of interest to your KB of interest. 

2. Provide a folder similar to the BC5CDR for the ontology you are interested in and train/test the model on this. You could conceivably produce this dataset somewhat programmatically using the MedMentions dataset since concepts in MedDRA are contained within UMLS. You'd have to arbitrate the cases where a single MedDRA concept is mapped to multiple UMLS concepts and vice-versa. 

Clearly, there's less overhead with option 1, but the domain shift between your KB and one of the original KBs may yield lower performance.
